src/msg/simple/Pipe.cc

```diff
@@ -854,10 +857,24 @@ void Pipe::set_socket_options()
     int r = -1;
 #ifdef IPTOS_CLASS_CS6
     int iptos = IPTOS_CLASS_CS6;
-    r = ::setsockopt(sd, IPPROTO_IP, IP_TOS, &iptos, sizeof(iptos));
-    if (r < 0) {
-      ldout(msgr->cct,0) << "couldn't set IP_TOS to " << iptos
-                         << ": " << cpp_strerror(errno) << dendl;
+
+    if (peer_addr.get_family() == AF_INET) {
+      r = ::setsockopt(sd, IPPROTO_IP, IP_TOS, &iptos, sizeof(iptos));
+      r = -errno;
```

Comment:
@tchaikov #14795 touches `src/msg/async/net_handler.cc`, not this file (`src/msg/simple/Pipe.cc`), also requires backporting the `set_priority` prototype which otherwise isn't in jewel.